frontend/src/features/admin-form/create/logic/components/LogicContent/InactiveLogicBlock/FieldLogicBadge.tsx

```diff
@@ -9,30 +10,73 @@ import { FormFieldWithQuestionNo } from '~features/form/types'
 import { LogicBadge } from './LogicBadge'
 
 interface FieldLogicBadgeProps {
-  field: FormFieldWithQuestionNo
+  field: FormFieldWithQuestionNo | null
+  errorType?: 'error' | 'info'
+  errorString?: string
 }
```

Comment:
`'info'` is not really an `errorType`, maybe `'variant'`? optionally you could also type this more strongly:

```typescript
type FieldLogicBadgeProps = 
  | { type: 'info', field: FormFieldWithQuestionNo }
  | { type: 'error', message: string }
}
```

Reply:
in fact if you look at it this way, the props required for the two cases are completely different, which is forcing us to use multiple `switch` statements and ternary operators all over the place. can consider splitting this into separate components. although it will result in a couple of lines of repeated code, it will reduce the case splits required everywhere else.

arguably in future we might need more types of `LogicBadge` and at some point it might make sense to refactor them into the same component, but for now imo these make more sense separately. up to you though!

Reply:
as discussed, `'error' | 'info'` are the default fallbacks when `field` is `null`. This type is not ideal but I think any other way introduces a lot of complexity into the code. For example, another possibility is that we can have another component that deals with error cases specifically (so `FieldLogicBadge` only takes in a field, and `ErrorLogicBadge` takes in the `{ variant: 'error' | 'info', message: string }`) but then the error-handling behavior is repeated everywhere you want a logic badge anyway, so it boils down to the same thing

Reply:
i'll change the names to `variant` and `message` though so its less confusing!